clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -4284,6 +4284,9 @@ bool SemaHLSL::transformInitList(const InitializedEntity &Entity,
   }
   size_t ExpectedSize = ILT.DestTypes.size();
   size_t ActualSize = ILT.ArgExprs.size();
+  if (ExpectedSize == 0 && ActualSize == 0)
```

Comment:
So this check is a little troubling b/c we have a divide by `ExpectedSize` not to far below here and based on this condition `ExpectedSize` can be `0` and it is undefined behavior to divide by zero.

So if there are more conditions here not represented in the code we should add assertions to document the invariants. 

CC @llvm-beanz

Reply:
In the case the object being initialized is an Incomplete Array Type, which afaict is the only place there is a division by 'ExpectedSize', 'ExpectedSize' should be at least 1. I think we can add an assertion that ExpectedSize will not be zero at that point.